### ISSUE TITLE
[Feature]: Add clock skew detection for Jamf Pro inventory failures

### ISSUE THREAD

**Issue by dan-snelson** [dan-snelson/Mac-Health-Check#101]
## What problem are you trying to solve?

Jamf Pro inventory submission can fail when the Mac's local clock differs too much from the Jamf Pro server time. I observed this in a Jamf Pro policy log:

```log
Clock Skew Error - There is a discrepancy between the local time and the JSS server time.
```

When this happens, Mac Health Check can surface Jamf inventory problems, but it does not currently diagnose local clock skew as a likely root cause.

## Proposed behavior

Add a health check that detects meaningful clock skew before Jamf Pro inventory submission, using `sntp` against a trusted time source such as `time.apple.com`.

The check should report compliant when the offset is within Jamf Pro tolerance, and action required when the offset exceeds 5 minutes. The result should include concise remediation guidance to verify automatic time sync and re-sync the Mac clock.

Suggested diagnostic commands:

```bash
date
sntp time.apple.com
sudo systemsetup -getnetworktimeserver
sudo systemsetup -getusingnetworktime
```

Suggested remediation commands:

```bash
sudo sntp -sS time.apple.com
sudo killall -HUP timed
```

## Alternatives you have considered

Mac Admins can diagnose this manually from the Jamf Pro policy log by running `date`, `sntp time.apple.com`, and `systemsetup` commands. That works for support staff, but it does not help end users or proactively identify the condition before Jamf inventory fails.

## Anything else

Jamf Pro typically tolerates up to 5 minutes of skew. If implemented, this should stay MDM-agnostic where practical because clock skew can affect more than Jamf Pro, but the user-visible motivation is Jamf Pro inventory failure.

**Comment by dan-snelson** [comment]
For our use-case, the error message would be _close_ to the following:

```
dialogUpdate "listitem: index: ${1}, icon: SF=$(printf "%02d" $(($1+1))).circle.fill weight=bold colour=${statusColorFail}, iconalpha: 1, subtitle: Please run macOS Maintenance from the ${organizationSelfServiceMarketingName}, status: fail, statustext: Skew ${clockSkew}"
```

**Comment by dan-snelson** [comment]
Addressed in `4.2.0b2`
